content/docs/reference/state.md

```diff
@@ -1,140 +1,307 @@
+##### Amazon S3
 
-To use the web backend, just run `pulumi login` again, and you’ll be back to using app.pulumi.com. If you have any
-existing stacks, however, you'll need to migrate them. It's easiest to just plan on recreating them.
+```
+pulumi login --cloud-url s3://my-pulumi-state-bucket
+```
 
-If you'd like to migrate your stacks from the filesystem to web, however, you can do the following. Suppose the stack
-"my-app-production" has been managed with a local checkpoint file, and you want to migrate it to pulumi.com. If you are
-currently logged in to the local endpoint, run the following commands:
+##### Google Cloud Storage
+
+```
+pulumi login --cloud-url gs://my-pulumi-state-bucket
+```
+
+##### Azure Blob Storage
+
+```
+pulumi login --cloud-url az://my-pulumi-state-bucket
+```
+
+This stores all stack checkpoints as JSON files to the `.pulumi` directory of
+your specified cloud URL.
+
+To control where these checkpoints get stored, refer to your cloud storage
+provider's documentation. See [Self-managed backend](#self-managed-backend) 
+for quick links to Amazon, Google, and Microsoft Azure's storage service quickstarts.
+
+You may omit `--cloud-url` or `-c` when logging in to a remote backend and just
+use `pulumi login s3://my-pulumi-state-bucket`.
+
+The precise JSON format these checkpoint files use is not documented, but is
+defined in the [APIType source code](
+https://github.com/pulumi/pulumi/blob/master/pkg/apitype/) if you'd like to
+understand their contents. Note that this is the same JSON format used by the
+`pulumi stack export` and `pulumi stack import` commands.
+
+#### Notes on self-managed backends
+
+If you lose the checkpoint for your stack, Pulumi will be unable to manage any
+existing resources. Additionally, since Pulumi thinks your stack is empty,
+Pulumi will attempt to recreate all of the resources in your stack the next
+time you run `pulumi up`.
+
+Some commands may behave slightly differently when using the local or remote
+storage endpoint. For example, when connected to `pulumi.com`, `pulumi up`
+ensures there are no other updates in flight for a given stack. This doesn't
+happen with self-managed backends. Pulumi also manages secrets using a key
+encrypted with a passphrase and stored in `Pulumi.<stack-name>.yaml`. This
+requires you enter the passphrase when you `preview`, `update`, or `delete`
+your stack. If you want to collaborate with another person, you'll need to
+share this passphrase with them as well. All of these overhead tasks will have
+to be managed separately when you opt into the local or remote state backend.
+
+### Going back to the pulumi.com backend
+
+So you've tried a self-managed backend and want to switch to the `pulumi.com`
+backend? This scenario can happen once users realize the benefit from the
+features that the `pulumi.com` backend delivers, particularly when
+operationalizing their usage.
```

Comment:
i really don't liek the editorializing of "once users realize the benefit".  It makes it sound (to me) that we think users who do things themselves are dumb and haven't realized what all the smart users have figured out.   I would make it much more encouraging.  Something like "If you are currently using a self-managed backend, but would now prefer to get simplify things by using the `pulumi.com` backend, you can do that by blah blah blah"

Reply:
Thanks, @CyrusNajmabadi . I've addressed all your comments -- can you please take a final look and see if all looks good?